java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -165,18 +166,40 @@ public void registerSSHMinion(String minionId, Optional<Long> proxyId,
     private void registerMinion(String minionId, boolean isSaltSSH,
                                 Optional<Long> saltSSHProxyId,
                                 Optional<String> activationKeyOverride) {
+
         // Match minions via their machine id
-        Optional<String> optMachineId = SALT_SERVICE.getMachineId(minionId);
-        if (!optMachineId.isPresent()) {
-            LOG.info("Cannot find machine id for minion: " + minionId);
-            return;
-        }
-        //FIXME: refactor this whole function so we don't have to call get
-        //in so many places.
-        String machineId = optMachineId.get();
-        MinionServer minionServer;
-        Optional<User> creator = MinionPendingRegistrationService.getCreator(minionId);
+        Opt.fold(SALT_SERVICE.getMachineId(minionId),
+            () -> {
+                LOG.info("Cannot find machine id for minion: " + minionId);
+                return false;
```

Comment:
I wonder why we log only with 'info' severity. We abort the registration so I think this is an error, isn't it?

It was this way in the original code already, so we might want to research a bit before changing this.
Maybe it has a reason i cannot see.

Reply:
I also think it should be logged as warning or error. Maybe warning if its somehow "normal" to have registration attempts without a valid machine id and we don't want to alarm the users, otherwise error. @renner any thoughts?